app/javascript/components/widget-transition-controls.test.jsx

```diff
@@ -54,6 +81,16 @@ describe('shouldRequestPrefetch', () => {
       }),
     ).toEqual(true);
   });
+
+  it('returns true, msRemaining as null will still trigger request prefetch (possible error from editing useEffect execution stop)', () => {
+    expect(
+      selectShouldRequestPrefetch({
+        msRemainingSinceLastStart: null,
+        startTimestamp: Date.now() - 21000,
+        prefetchRequestedYet: false,
+      }),
+    ).toEqual(true);
```

Comment:
I disagree with this assertion. I think `selectShouldRequestPrefetch` should either return false (or throw an exception) if either `msRemainingSinceLastStart` or `startTimestamp` are null. Both are an invalid state for calling this function.

Reply:
I think adding the null checking/throwing and this test should be a separate commit that precedes your feature commit. Something like:
* refactor: rename the default twitter export
* fix: add isStarted check to prevent incorrectly calling `selectShouldRequestPrefetch` with a null value
* fix: disallow calling `selectShouldRequestPrefetch` with null `msRemaining`/`startTimestamp`
* feat: start the sidebar timer after the widget's data is loaded